File: zlcore/src/main/java/com/zaitunlabs/zlcore/views/ASGestureListener.java
package com.zaitunlabs.zlcore.views;

public interface ASGestureListener {
	public boolean deltaMoveInsideParameter(int swipeType, float x, float y, float dx, float dy, float fromDownDX, float fromDownDY);
	public boolean deltaMoveOutsideParameter(int swipeType, float x, float y, float dx, float dy, float fromDownDX, float fromDownDY);
	public boolean movingSpeed(float xSpeed, float ySpeed);
	public boolean upEventOccurred(float x, float y);
	public boolean downEventOccured(float x, float y);
	public boolean cancelEventOccured(float x, float y);
	public boolean clickEventOccured();
	public boolean longClickEventOccured();
	public boolean doubleTapEventOccured();
	public boolean swipeEventOccured(int swipeType, float x, float y, float dx, float dy);
	public boolean swipeTypeFinal(int swipeType);
	
	//property
	public boolean isClickEnabled();
	public boolean isLongClickEnabled();
	public boolean isDoubleTapEnabled();
	public boolean isSwipeEnabled();
}

File: zlcore/src/main/java/com/zaitunlabs/zlcore/views/ASGestureListener.kt
package com.zaitunlabs.zlcore.views

interface ASGestureListener {

    //property
    val isClickEnabled: Boolean
    val isLongClickEnabled: Boolean
    val isDoubleTapEnabled: Boolean
    val isSwipeEnabled: Boolean
    fun deltaMoveInsideParameter(swipeType: Int, x: Float, y: Float, dx: Float, dy: Float, fromDownDX: Float, fromDownDY: Float): Boolean
    fun deltaMoveOutsideParameter(swipeType: Int, x: Float, y: Float, dx: Float, dy: Float, fromDownDX: Float, fromDownDY: Float): Boolean
    fun movingSpeed(xSpeed: Float, ySpeed: Float): Boolean
    fun upEventOccurred(x: Float, y: Float): Boolean
    fun downEventOccured(x: Float, y: Float): Boolean
    fun cancelEventOccured(x: Float, y: Float): Boolean
    fun clickEventOccured(): Boolean
    fun longClickEventOccured(): Boolean
    fun doubleTapEventOccured(): Boolean
    fun swipeEventOccured(swipeType: Int, x: Float, y: Float, dx: Float, dy: Float): Boolean
    fun swipeTypeFinal(swipeType: Int): Boolean
}
